packages/babel-parser/test/fixtures/experimental/decorators/invalid-decorator-array-pattern/input.js

```diff
@@ -0,0 +1 @@
+var [@foo foo ] = foo;
```

Comment:
For stage 3 decorators, currently Babel throws a recoverable error message "Decorators cannot be used to decorate parameters.", which is quite confusing. Here we throw an unrecoverable error since we already do that for object patterns.